### PR TITLE
Bluetooth: audio: ascs: Do not store Bluetooth address

### PR DESCRIPTION
**Pull request by MariuszSkamra**
The struct bt_ascs holds the conn reference already, thus it's not needed to store the device address.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -873,35 +844,37 @@ static void ase_disable(struct bt_ascs_ase *ase)
...
-		ascs->conn = NULL;
+		if (stream != NULL && stream->conn != NULL) {
+			bt_conn_unref(stream->conn);
+			stream->conn = NULL;
+		}
+	}
 
-		for (size_t i = 0U; i < ARRAY_SIZE(ascs->ases); i++) {
-			struct bt_audio_stream *stream = ascs->ases[i].ep.stream;
+	for (size_t i = 0; i < ASE_COUNT; i++) {
```
Comment (sjanc) [inline subsys/bluetooth/audio/ascs.c:865]: both loops are iterating over same things,  is this on purpose?    (btw I like use of array_size better for this purpose)

**Review by Thalley — DISMISSED** [review]
Good catch. After we removed the (flawed) implementation of caching configurations, we no longer need to store the address. 

If we want to be able to cache the codec config again, we would need to revert this change again though, but let's cross that bridge if we get to it.

**Comment by MariuszSkamra** [comment]
> Good catch. After we removed the (flawed) implementation of caching configurations, we no longer need to store the address.
> 
> If we want to be able to cache the codec config again, we would need to revert this change again though, but let's cross that bridge if we get to it.

We could consider providing an API to application that could cache the device. I guess it's better to cache and restore the device from flash instead that will preserve even if device is rebooted (e.g. battery replacement)

**Review by sjanc — DISMISSED** [review]

**Comment by MariuszSkamra** [comment]
Rebased

**Review by Thalley — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**